Sluggo/Models/User.swift

```diff
@@ -14,4 +14,42 @@ struct UserRecord: Codable {
     var first_name: String?
     var last_name: String?
     var username: String
+    
+    init(id: Int, email: String, first_name: String?, last_name: String?, username: String) {
+        self.id = id
+        self.email = email
+        self.first_name = first_name
+        self.last_name = last_name
+        self.username = username
+    }
+    
+    init?(fromJSONString: String) { // Initializer for data from JSON
+        // Get data
+        guard let jsonData = fromJSONString.data(using: .utf8) else { return nil }
+        
+        // Attempt decoding
+        let decoder = JSONDecoder()
+        decoder.dateDecodingStrategy = .iso8601
+        
+        guard let userRecord = try? decoder.decode(UserRecord.self, from: jsonData) else { return nil }
+        
+        // Initialize using standard initializer
+        self.init(id: userRecord.id, email: userRecord.email, first_name: userRecord.first_name, last_name: userRecord.last_name, username: userRecord.username)
```

Comment:
what's userRecord's type? You could probably abstract most of the json decoding to a base class and then define the initializer to build from a json object